pybleau/app/ui/filter_expression_editor.py

```diff
@@ -1,31 +1,285 @@
+    #: Class to use to create TraitsUI window to open controls
+    view_klass = Any(default_value=View)
+
+    #: Name for the VGroup
+    group_label = Str
+
+    #: Internal list of trait names
+    _trait_names = List
+
+    def __init__(self, traits_and_names, handler, group_label="", **traits):
+        super(ScrollableButtonListView, self).__init__(**traits)
+        self.traits_and_names = traits_and_names
+        self.handler = handler
+        self.group_label = group_label
+        self._make_trait_names()
+
+    def _make_trait_names(self):
+        # add the trait names to this object, and connect the handler function
+        for trait_name, label in self.traits_and_names.items():
+            self._trait_names.append(trait_name)
+            self.add_trait(trait_name, Button(label))
+            self.observe(self.handler, trait_name)
+
+    def traits_view(self):
+        items = [Item(n, show_label=False) for n in self._trait_names]
+
+        return self.view_klass(
+            VGroup(*items, label=self.group_label),
+            scrollable=True
+        )
 
 
 class FilterExpressionEditorView(HasStrictTraits):
     """ Provides a resizeable edit view of the filter string (or any string).
     """
 
     #: Expression representing the dataframe filter
-    exp = Str
+    expr = Str
+
+    #: Dataframe upon which filter will operate
+    source_df = Instance(DataFrame)
 
     #: Class to use to create TraitsUI window to open controls
     view_klass = Any(View)
 
+    #: Selected columns which should have their own tab
+    included_cols = List
+
+    #: Status bar for the window
+    status_bar = Instance(StatusBarManager)
+
+    #: Message that is displayed in the status bar
+    status_bar_message = Str
+
+    #: Dictionary of translated trait names with original names as keys
+    traited_names = Dict
+
+    #: Dictionary of original names with translated trait names as keys
+    original_names = Dict
+
+    #: List of column names that should have their own tabs
+    included_col_values = Dict
+
+    #: Boolean that determines whether the clicked value is copied or appended
+    auto_append = Bool(True)
+
+    #: Enum that allows selection of action that occurs when clicking
+    click_type = Enum(values="_click_action_options")
+
+    #: flag describing the initialization status of this instance
+    _edit_initialized = Bool(False)
+
+    #: List of actions that can occur when clicking a button
+    _click_action_options = List([APPEND_TO_FILTER, COPY_TO_CLIPBOARD])
+
+    #: Provider of a scrollable view for the column name buttons
+    _scrollable_column_names = Instance(ScrollableButtonListView)
+
+    #: List of scroll list views for each column in included_cols
+    _included_col_scroll_list = List
+
+    def __init__(self, source_df: DataFrame, **traits):
+        super(FilterExpressionEditorView, self).__init__(**traits)
+        if not any(sorted(source_df)):
+            msg = f"The source_df contains no columns"
```

Comment:
The `f` can be removed. It's unnecessary and seems to break flake8 test in appveyor: https://ci.appveyor.com/project/jonathanrocher/pybleau/builds/36405932

Reply:
Will do.